### PR TITLE
dts: nrf5340_cpuapp: add cryptocell node

### LINKED ISSUE #25493: devicetree: nRF5340 application core DTSI is missing cryptocell node
**Describe the bug**

The nRF5340 SoC's application core [has a cryptocell peripheral](https://infocenter.nordicsemi.com/index.jsp?topic=%2Fps_nrf5340%2Fkeyfeatures_html5.html), but this is not available in the corresponding DTSI.

Now that entropy sources have moved to DTS's zephyr,entropy chosen node from Kconfig's CONFIG_ENTROPY_NAME, this means users can no longer easily instantiate devices for managing the cryptocell in a way that interoperates with the rest of the system.

**Expected behavior**

nrf5340_cpuapp.dtsi should have nodes for all hardware addressable from the secure world.

**Impact**

Showstopper for "standard zephyr" entropy use on this core.

**Environment (please complete the following information):**
 - Commit SHA or Version used: v2.3.0-rc1 for example

### PR DESCRIPTION
**Pull request by mbolivar-nordic**
Add the missing node. This is a secure-mapped peripheral, so do not
add it to the common include file nrf5340_cpuapp_common.dtsi.

Fixes: #25493

### CODE REVIEW

**Comment by mbolivar-nordic** [comment]
Do not merge without @frkv approval

**Review by galak — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Review by ioannisg — DISMISSED** [review]
shouldn't this be 312?

File: dts/arm/nordic/nrf5340_cpuapp.dtsi
Comment (frkv) [inline dts/arm/nordic/nrf5340_cpuapp.dtsi:102]: Interrupt is 68, not 64

**Review by frkv — APPROVED** [review]

**Comment by galak** [comment]
Seems like you now need a binding for `arm,cryptocell-312` & `nordic,nrf-cc312`

**Comment by mbolivar-nordic** [comment]
> Seems like you now need a binding for `arm,cryptocell-312` & `nordic,nrf-cc312`

yep, done now

**Review by anangl — APPROVED** [review]

**Comment by mbolivar-nordic** [comment]
@ioannisg care to take another look?

**Pull request merged**
